com.unity.shadergraph/Editor/Drawing/MaterialGraphEditWindow.cs

```diff
@@ -251,6 +261,17 @@ void Update()
                 // Called again to handle changes from deserialization in case an undo/redo was performed
                 graphEditorView.HandleGraphChanges(wasUndoRedoPerformed);
                 graphObject.graph.ClearChanges();
+
+                if (restoreNodeSelectionState)
+                {
+                    foreach (var nodeView in graphEditorView.graphView.graphElements.ToList().OfType<MaterialNodeView>())
+                    {
+                        if (nodeView != null && selectionUids.Contains(nodeView.node.objectId))
+                        {
+                            graphEditorView.graphView.AddToSelection(nodeView);
+                        }
+                    }
+                }
```

Comment:
Basic cache/restore selection step. The selection state gets manipulated in quite a few places as many of the 'changes' procedures are called in duplicity. This solution should probable live in MaterialGraphView instead, but it's execution flow for undo/redo needs a bit more work to support that.